Can't "make install" on read-only source directory

When the build directory is mounted via NFS one can't perform a "make install".

**Environment**
My build machine is running Fedora 43.  The base "openscad" directory is in a directrory mounted from a NAS so that the same sources can be used for multiple build machines.  A "cmake -B build-fed43" was done for this system, other subdirectories are used for other targets.

**Issue**
The openscad executable can be built as a non-root user but when I try to install it fails.  I think this is because root does not have rights to write to the share:

```
cd build-fed43
make preinstall       # done to minimize the root owned files, I hope
sudo make install
```

It fails with:


```
...
[ 97%] Built target OpenSCADUnitTests
[ 98%] Generating issue2342.scad
/bin/sh: line 1: /home/gaa/OpenSCAD/openscad/tests/data/scad/issues/issue2342.scad: Permission denied
make[2]: *** [tests/CMakeFiles/issue2342.dir/build.make:71: tests/CMakeFiles/issue2342] Error 1
make[1]: *** [CMakeFiles/Makefile2:1559: tests/CMakeFiles/issue2342.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```


I notice that every time a make is run, the issue2342.scad file is recreated.  Perhaps it should not be touched again just to install.  If every make changes "source", then it would seem that one can't share the source tree via NFS (or SMB).

These are the files that I found get touched every make, even if already made:

```
tests/data/svg/viewbox/viewbox_600x600_slice_xMinYMin.svg
tests/data/svg/viewbox/viewbox_300x400_meet_xMinYMin.svg
tests/data/svg/viewbox/viewbox_600x200_meet_xMinYMin.svg
tests/data/svg/viewbox/viewbox_600x200_slice_xMidYMin.svg
tests/data/svg/viewbox/viewbox_300x400_none.svg
tests/data/svg/viewbox/viewbox_600x200_slice_xMaxYMin.svg
tests/data/svg/viewbox/viewbox_600x200_slice_xMinYMin.svg
tests/data/svg/viewbox/viewbox_300x400_meet_xMidYMin.svg
tests/data/svg/viewbox/viewbox_600x200_meet_xMinYMax.svg
tests/data/svg/viewbox/viewbox_600x200_meet_xMinYMid.svg
tests/data/svg/viewbox/viewbox_300x400_meet_xMaxYMin.svg
tests/data/svg/viewbox/viewbox_600x600_slice_xMinYMid.svg
tests/data/svg/viewbox/viewbox_600x600_slice_xMinYMax.svg
tests/data/svg/viewbox/viewbox_600x200_none.svg
tests/data/scad/issues/issue2342.scad
```

--- a/.gitignore
+++ b/.gitignore
@@ -93,10 +93,8 @@ tests/data/python/gen_svg_viewbox_tests.py
 tests/data/scad/2D/features/import_dxf-tests.scad
 tests/data/scad/3D/features/import_stl-tests.scad
 tests/data/scad/3D/features/import_3mf-tests.scad
-tests/data/scad/issues/issue2342.scad
 tests/data/scad/misc/include-tests.scad
 tests/data/scad/misc/use-tests.scad
-tests/data/svg/viewbox/viewbox_*.svg
 tests/regression/pdfexporttest/centered.pdf
 tests/regression/pdfexporttest/simple-pdf.pdf
 tests/regression/export-pdf-fill/centered.pdf

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -178,13 +178,19 @@ configure_file(${TEST_PYTHON_DIR}/gen_svg_viewbox_tests-template.py
 
 # generate a very large scad file which we would rather not commit to the source tree
 # this is for stress-testing the parser
+# Generated into the build dir (CCBD) so the source tree can stay read-only (#6700)
+file(MAKE_DIRECTORY ${CCBD}/data/scad/issues)
 add_custom_target(issue2342 ALL
-  COMMAND ${Python3_EXECUTABLE} ${TEST_PYTHON_DIR}/gen_issue2342.py ">${TEST_SCAD_DIR}/issues/issue2342.scad"
+  COMMAND ${Python3_EXECUTABLE} ${TEST_PYTHON_DIR}/gen_issue2342.py ">${CCBD}/data/scad/issues/issue2342.scad"
   WORKING_DIRECTORY ${GEN_SCRIPT_DIR}
   COMMENT "Generating issue2342.scad"
 )
+file(MAKE_DIRECTORY ${CCBD}/data/svg/viewbox)
+file(MAKE_DIRECTORY ${CCBD}/data/scad/svg/extruded)
+configure_file(${TEST_DATA_DIR}/svg/viewbox/viewbox-tests.svg.in ${CCBD}/data/svg/viewbox/viewbox-tests.svg.in COPYONLY)
+configure_file(${TEST_SCAD_DIR}/svg/extruded/viewbox-test.scad ${CCBD}/data/scad/svg/extruded/viewbox-test.scad COPYONLY)
 add_custom_target(svg_viewbox_tests ALL
-  COMMAND ${Python3_EXECUTABLE} ${TEST_PYTHON_DIR}/gen_svg_viewbox_tests.py "${TEST_DATA_DIR}/svg/viewbox" "${TEST_SCAD_DIR}/svg/extruded"
+  COMMAND ${Python3_EXECUTABLE} ${TEST_PYTHON_DIR}/gen_svg_viewbox_tests.py "${CCBD}/data/svg/viewbox" "${CCBD}/data/scad/svg/extruded"
   WORKING_DIRECTORY ${GEN_SCRIPT_DIR}
   COMMENT "Generating svg viewbox tests"
 )
@@ -373,7 +379,7 @@ list(APPEND ECHO_FILES ${FUNCTION_FILES} ${MISC_FILES} ${REDEFINITION_FILES}
   ${TEST_SCAD_DIR}/issues/issue1923.scad
   ${TEST_SCAD_DIR}/misc/preview_variable.scad
   ${TEST_SCAD_DIR}/issues/issue1851-each-fail-on-scalar.scad
-  ${TEST_SCAD_DIR}/issues/issue2342.scad
+  ${CCBD}/data/scad/issues/issue2342.scad
   ${TEST_SCAD_DIR}/issues/issue3118-recur-limit.scad
   ${TEST_SCAD_DIR}/issues/issue3541.scad
   ${TEST_SCAD_DIR}/misc/function-scope.scad
@@ -864,9 +870,9 @@ endif()
 add_cmdline_test(throwntogether-cgal  OPENSCAD FILES ${ALL_THROWNTOGETHER_FILES} SUFFIX png EXPECTEDDIR throwntogether ARGS --preview=throwntogether --backend=cgal)
 add_cmdline_test(throwntogether-manifold  OPENSCAD FILES ${ALL_THROWNTOGETHER_FILES} SUFFIX png EXPECTEDDIR throwntogether ARGS --preview=throwntogether --backend=manifold)
 
-set(VIEWBOX_TEST "${TEST_SCAD_DIR}/svg/extruded/viewbox-test.scad")
+set(VIEWBOX_TEST "${CCBD}/data/scad/svg/extruded/viewbox-test.scad")
 foreach(TEST ${SVG_VIEWBOX_TESTS})
-  add_cmdline_test(svgviewbox-${TEST} OPENSCAD ARGS --imgsize 600,600 "-Dfile=\"${TEST_DATA_DIR}/svg/viewbox/${TEST}.svg\";" SUFFIX png FILES ${VIEWBOX_TEST})
+  add_cmdline_test(svgviewbox-${TEST} OPENSCAD ARGS --imgsize 600,600 "-Dfile=\"${CCBD}/data/svg/viewbox/${TEST}.svg\";" SUFFIX png FILES ${VIEWBOX_TEST})
 endforeach()
 
 add_cmdline_test(svgimport OPENSCAD ARGS --imgsize 600,600 SUFFIX png FILES

